tools/ci_connector_ops/ci_connector_ops/pipelines/contexts.py

```diff
@@ -126,7 +131,6 @@ def report(self) -> Report:  # noqa D102
     @report.setter
     def report(self, report: Report):  # noqa D102
         self._report = report
-        self.state = ContextState.SUCCESSFUL if report.success else ContextState.FAILURE
```

Comment:
This was the root cause of the false positive errors: 
When we encounter an unhandled error an empty report is generated. 
Report is determined as successful when it has no failed step. 
When a report is empty it has no failed step...
I preferred to explicitly determine the final state of a context with a method I implemented below.